### DISCUSSION TITLE
Services (Itils definition) in NetBox

### DISCUSSION THREAD

**Discussion by tomhur** [netbox-community/netbox#6689] [General]
In my company we have a need of defining what devices, VM's and whatnot helps helps delivering a service, as ITIL defines a service:
> Service - A definition according to ITIL®: "A means of delivering value to customers by facilitating outcomes customers want to achieve without the ownership of specific costs and risks. The term ‘service’ is sometimes used as a synonym for core service, IT service or service package." [1][1]

We have started to use Device roles for this and it is working good, for the most part. We've added a few custom fields
![image](https://user-images.githubusercontent.com/19687175/124129390-0801c400-da7e-11eb-9ab0-898abbfbe345.png)
to store the extra data we want and I'm looking forward to [Add Device role parent/child support like regions #6290](https://github.com/netbox-community/netbox/issues/6290) that I think will help us organise things.

What I think I would want is the possibility to add several device roles to a singe device or VM in order to help visualize it's part of delivering multiple services. Also some kind of dependency tree, some services are dependent of other services.

I'm wondering, is there someone else trying to deal with the same problem as I do and how do you solve it? Is this out of scope for NetBox? If I'm able to convince my company to contribute to NetBox in this area, is it something others would want or would it just bloat NetBox?

[1]: https://wiki.en.it-processmaps.com/index.php/ITIL_Glossary/_ITIL_Terms_S#Service

**Comment by candlerb** [comment]
> What I think I would want is the possibility to add several device roles to a singe device or VM in order to help visualize it's part of delivering multiple services.

"Tags" are probably the best way to model that in Netbox right now.  However, [custom fields for tags](https://github.com/netbox-community/netbox/issues/5116) have been soundly rejected.  And they don't form a tree (there are only a few places in Netbox which form trees, like Region and Location).

It sounds to me like your best approach would be to write a Netbox [plugin](https://netbox.readthedocs.io/en/stable/plugins/).

**Reply by tomhur** [reply]
Yes, looked at how plugins work and it does not seem like a hard problem to solve.

Thank you!
